strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/providers/io/RepositoryFiles.java

```diff
@@ -0,0 +1,72 @@
+package org.carlspring.strongbox.providers.io;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Set;
+import java.util.StringJoiner;
+import java.util.stream.Collectors;
+
+import org.carlspring.strongbox.artifact.coordinates.ArtifactCoordinates;
+
+public abstract class RepositoryFiles
+{
+
+    public static Boolean isChecksum(RepositoryPath path)
+        throws IOException
+    {
+        return (Boolean) Files.getAttribute(path, formatAttributes(RepositoryFileAttributeType.CHECKSUM));
+    }
+
+    public static Boolean isMetadata(RepositoryPath path)
+        throws IOException
+    {
+        return (Boolean) Files.getAttribute(path, formatAttributes(RepositoryFileAttributeType.METADATA));
+    }
+
+    public static Boolean isArtifact(RepositoryPath path)
+        throws IOException
+    {
+        return (Boolean) Files.getAttribute(path, formatAttributes(RepositoryFileAttributeType.ARTIFACT));
+    }
+
+    public static ArtifactCoordinates readCoordinates(RepositoryPath path)
+        throws IOException
+    {
+        return (ArtifactCoordinates) Files.getAttribute(path,
+                                                        formatAttributes(RepositoryFileAttributeType.COORDINATES));
+    }
+
+    public static String formatAttributes(RepositoryFileAttributeType... attributeTypes)
```

Comment:
Would you mind explaining this method, please? (Just so that I could get a better understanding of what's going on... For example, what's the `strongbox:*` part for, etc). Thanks! :)

Reply:
this is according nio.file specification about file attributes. There is `BasicFileAttributes` which is default. And also there can be other attributes which need to be prefixed with some keyword. For exemple there is `PosixFileAttributes` which is prefixed with `posix:`. And for strongbox we have `RepositoryFileAttributes` which is prefixed with `strongbox:`. You can note `java.nio.file.Files.readAttributes` javadoc for this.

Reply:
Right, now I think I get it. Thanks! :)